scrypto/src/types/nft_key.rs

```diff
@@ -0,0 +1,34 @@
+use core::fmt::{Display, Formatter};
+use core::fmt;
+use crate::rust::vec;
+use crate::rust::vec::Vec;
+use sbor::*;
+use crate::buffer::{SCRYPTO_TYPE_NFT_KEY};
+
+/// Represents a key for an NFT resource
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Encode, Decode, Describe)]
+pub struct NftKey(Vec<u8>);
+
+impl NftKey {
+    pub fn new(v: Vec<u8>) -> Self {
+        NftKey(v)
+    }
+}
+
+impl From<u128> for NftKey {
+    fn from(u: u128) -> Self {
+        NftKey(u.to_be_bytes().to_vec())
+    }
+}
```

Comment:
I'd suggest we add a conversion for any type with `sbor::Encode` trait.

Reply:
This interface should only support conversions to bytes array of bounded length. `sbor::Encode` would expose too much.